server/lib/util.py

```diff
@@ -500,3 +503,89 @@ def gzip_compress_response(raw_content, is_json):
   if is_json:
     response.headers['Content-Type'] = 'application/json'
   return response
+
+
+def fetch_highest_coverage(parent_entity: str,
+                           child_type: str,
+                           variables: List[str],
+                           all_facets: bool,
+                           facet_ids: List[str] = None):
+  """
+  Fetches the latest available data with the best coverage for the given a
+  parent entity, child type, variables, and facets
+
+  Response format:
+  {
+    "facets": {
+      <facet_id>: {<facet object>}
+    },
+    "data": {
+      <var_dcid>: {
+        <entity_dcid>: {
+          <observation point(s)>
+        }
+      }
+    }
+  }
+  """
+  # Heuristic for fetching "latest data with highest coverage":
+  # Choose the date with the most data coverage from either:
+  # (1) last 5 observation dates
+  # (2) 5 years from the most recent observation date
+  # whichever set is greater
+  FETCH_HIGHEST_COVERAGE_FROM_DATES = 5
```

Comment:
nit: these constant names are slightly confusing, maybe just 'MAX_DATES_TO_CHECK' and 'MAX_YEARS_TO_CHECK'?

Reply:
Good callout-updated